### PR TITLE
feat: support dynamic and configurable robots.txt

### PR DESCRIPTION
**Pull request by dominique-pfister**
To replace the VCL in aem.page/live to return the internal `robots.txt` on preview, inner and outer CDN

### CODE REVIEW

**Review by tripodsan — CHANGES_REQUESTED** [review]
probably also add the config specific surrogate key:
https://github.com/adobe/helix-html-pipeline/blob/ee0e9f7df20609c40792add733b05944ab930dc4/src/json-pipe.js#L94

File: src/robots-pipe.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      'content-type': 'text/plain; charset=utf-8',
+    },
+  });
+}
+
+async function computeSurrogateKeys(state) {
+  const keys = [];
+
+  const pathKey = `${state.ref}--${state.repo}--${state.owner}${state.info.path}`;
+  keys.push(await computeSurrogateKey(`${state.site}--${state.org}_config.json`));
```
Comment (dominique-pfister) [inline src/robots-pipe.js:42]: added own method to include that key

File: src/robots-pipe.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+ *
+ * @param {import('./PipelineRequest.js').PipelineRequest} req request
+ * @returns {Array<String>} array of hosts
+ */
+function getForwardedHosts(req) {
+  const xfh = req.headers.get('x-forwarded-host');
+  if (!xfh) {
+    return [];
+  }
+  return xfh.split(',').map((v) => v.trim());
```
Comment (tripodsan) [inline src/robots-pipe.js:85]: filter empty segmetents

File: src/robots-pipe.js [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    },
+  });
+
+  const { partition } = state;
+  const forwardedHosts = getForwardedHosts(req);
+
+  if (partition === 'preview' || forwardedHosts.every(
+    (host) => !EXCLUDED_HOSTS.includes(host)
+      && INTERNAL_DOMAINS.some((domain) => host.endsWith(domain)),
+  )) {
```
Comment (dominique-pfister) [inline src/robots-pipe.js:165]: return internal `robots.txt` on preview or when all listed hosts are internal (excluding `www.aem.live`)

File: test/robots-pipe.test.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+      vary: 'x-forwarded-host',
+      'x-surrogate-key': 'U_NW4adJU7Qazf-I ref--repo--owner_robots.txt ZcR1sjWODctSccZh',
+    });
+    assert.strictEqual(resp.body, `User-Agent: *
+Allow: /
+
+Sitemap: https://www.example.com/sitemap.xml`);
+  });
+
+  it('renders robots from live with www.aem.live', async () => {
```
Comment (dominique-pfister) [inline test/robots-pipe.test.js:180]: test to assert `www.aem.live` does not get treated as internal domain

**Review by stefan-guggisberg — APPROVED** [review]

**Review by tripodsan — APPROVED** [review]
I think it would suffice to check if the first xfh segment is not a internal host. but I think this works, too.

**Pull request merged**